appservice/src/deploy/waitForDeploymentToComplete.ts

```diff
@@ -149,7 +148,7 @@ async function tryGetLatestDeployment(context: IActionContext, kuduClient: KuduC
         // Use "latest" to get the deployment before we know the "initialReceivedTime" or "permanentId"
         try {
             deployment = await retryKuduCall(context, 'getResult', async () => {
-                return <DeployResult | undefined>await kuduClient.deployment.getResult('latest');
+                return await kuduClient.deployment.getResult('latest');
```

Comment:
Have you tried this when there's no deployment yet?  I know that use to cause an error, but the return type doesn't include "undefined" so I'm wondering what happens.

Reply:
You just mean double check the logic for the catch block? My change here (to remove the type casting) should have no functional effect. It's possible the error thrown is different (i.e. maybe errorType ends up being "NotFound" instead of "404"), but I double checked deploying to a fresh app and it seemed to work. This scenario should be covered by nightly tests and CTI tests, so I feel like trying it once was enough for now.